components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -1096,6 +1096,38 @@ public String getServiceProviderNameByClientId(String clientId, String clientTyp
 
     }
 
+    /**
+     * @param inboundKey
+     * @param inboundType
+     * @param tenantDomain
+     * @return
+     * @throws IdentityApplicationManagementException
+     */
+    @Override
+    public String getApplicationResourceIDByInboundKey(String inboundKey, String inboundType,
+                                                       String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        if (StringUtils.isEmpty(inboundKey) || StringUtils.isEmpty(inboundType) || StringUtils.isEmpty(tenantDomain)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error while retrieving resource id. One of inboundKey, inboundType or tenantDomain " +
+                        "parameters were found to be empty.");
+            }
+            return null;
+        }
+
+        ApplicationDAO appDAO = ApplicationMgtSystemConfig.getInstance().getApplicationDAO();
+        String resourceId =  appDAO.getApplicationResourceIDByInboundKey(inboundKey, inboundType, tenantDomain);
+
+        if (resourceId == null) {
```

Comment:
don't we need to throw exception here? An app can't be without an appId.

Reply:
what if the given inboundKey is incorrect? then resource id should be null right?